Sources/WalletConnectRelay/RelayClient.swift

```diff
@@ -156,24 +156,87 @@ public final class RelayClient {
     }
 
     public func subscribe(topic: String) async throws {
-        logger.debug("Subscribing to topic: \(topic)")
+        logger.debug("[Subscribe] Subscribing to topic: \(topic)")
+
         let rpc = Subscribe(params: .init(topic: topic))
-        let request = rpc
-            .asRPCRequest()
-        let message = try! request.asJSONEncodedString()
+        let request = rpc.asRPCRequest()
+        let message = try request.asJSONEncodedString()
+
         try await dispatcher.protectedSend(message)
-        observeSubscription(requestId: request.id!, topics: [topic])
+
+        // Wait for relay's subscription response
+        try await waitForSubscriptionResponse(
+            requestId: request.id!,
+            topics: [topic],
+            logPrefix: "[Subscribe]"
+        )
     }
 
     public func batchSubscribe(topics: [String]) async throws {
         guard !topics.isEmpty else { return }
-        logger.debug("Subscribing to topics: \(topics)")
+        logger.debug("[BatchSubscribe] Subscribing to topics: \(topics)")
+
         let rpc = BatchSubscribe(params: .init(topics: topics))
-        let request = rpc
-            .asRPCRequest()
-        let message = try! request.asJSONEncodedString()
+        let request = rpc.asRPCRequest()
+        let message = try request.asJSONEncodedString()
+
         try await dispatcher.protectedSend(message)
-        observeSubscription(requestId: request.id!, topics: topics)
+
+        // Same wait, but for multiple topics
+        try await waitForSubscriptionResponse(
+            requestId: request.id!,
+            topics: topics,
+            logPrefix: "[BatchSubscribe]"
+        )
+    }
+
+    private func waitForSubscriptionResponse(
+        requestId: RPCID,
+        topics: [String],
+        logPrefix: String
+    ) async throws {
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
+            var cancellable: AnyCancellable?
+
+            cancellable = subscriptionResponsePublisher
+                // Only handle responses matching this request ID
+                .filter { $0.0 == requestId }
+                // Convert Never to RelayError so we can throw on timeout
+                .setFailureType(to: RelayError.self)
+                // Enforce a 60-second timeout
+                .timeout(.seconds(60), scheduler: concurrentQueue, customError: { .requestTimeout })
```

Comment:
```suggestion
                // Enforce a 30-second timeout
                .timeout(.seconds(30), scheduler: concurrentQueue, customError: { .requestTimeout })
```

Don't think anybody would wait a minute - up to you